### PR TITLE
git add secondary architecture

### PR DESCRIPTION
**Pull request by Begasus**
git can't be build with gcc2 due to parse errors anymore

### CODE REVIEW

**Comment by Begasus** [comment]
Related to: https://github.com/haikuports/haikuports/pull/6807

**Comment by augiedoggie** [comment]
This might need a "conflicts" entry.  I was able to install `git_x86` while `git` was still installed.

**Comment by Begasus** [comment]
> This might need a "conflicts" entry. I was able to install `git_x86` while `git` was still installed.

My idea was to get rid of the "older" versions, I don't think there is no "need" to keep them around (once things are fine), working fine for so far here.
Only thing I can't get working right is git-gui (but I can't run current version either on 32bit/64bit).

**Comment by korli** [comment]
It should "replaces" the old git on x86_gcc2.

**Comment by Begasus** [comment]
> It should "replaces" the old git on x86_gcc2.

Should it use REPLACES only for the main package, or also on the sub-packages?

File: dev-vcs/git/git-2.36.1.recipe [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+	git$secondaryArchSuffix = $portVersion compat >= 2.3
+	cmd:git = $portVersion compat >= 2.3
+	cmd:git_receive_pack = $portVersion compat >= 2.3
+	cmd:git_shell = $portVersion compat >= 2.3
+	cmd:git_upload_archive = $portVersion compat >= 2.3
+	cmd:git_upload_pack = $portVersion compat >= 2.3
+	"
+REPLACES="
+	git
+	"
```
Comment (Begasus) [inline dev-vcs/git/git-2.36.1.recipe:42]: This seems to do the trick here @korli , checked with pkgman to install git and got a conflict with the installed git_x86

Comment (korli) [reply dev-vcs/git/git-2.36.1.recipe:42]: it's only for x86_gcc2. please add a if ...

**Comment by korli** [comment]
> Should it use REPLACES only for the main package, or also on the sub-packages?

for all.

File: dev-vcs/git/git-2.36.1.recipe (hunk 1 above)
Comment (Begasus) [reply dev-vcs/git/git-2.36.1.recipe:42]: Right, wouldn't work on x86_64 ;)

File: dev-vcs/git/git-2.36.1.recipe [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+	REPLACES_source="
+		git_source
+		"
+fi
+PROVIDES_svn="
+	git${secondaryArchSuffix}_svn = $portVersion compat >= 2.3
+	cmd:git_svn = $portVersion compat >= 2.3
+	"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	REPLACES_svn="
```
Comment (korli) [inline dev-vcs/git/git-2.36.1.recipe:105]: i think one could use a loop here:
for pkg in arch cvs daemon email gui source svn; do eval "REPLACES_${pkg}=git_${pkg}"; done

**Comment by Begasus** [comment]
Checked on 32bit, haikuporter mentions the REPLACES, checked on 64bit, no mention there, so gather latest changes are OK
As mentioned on IRC earlier, some people are able to run git-gui (gitk) fine, for me it still doesn't

File: dev-vcs/git/git-2.36.1.recipe (hunk 2 above)
Comment (Begasus) [reply dev-vcs/git/git-2.36.1.recipe]: That worked :)

**Comment by korli** [comment]
LGTM

**Comment by waddlesplash** [comment]
Not sure why you requested my review, @korli is just as or more competent here than I am :)

**Comment by Begasus** [comment]
> Not sure why you requested my review, @korli is just as or more competent here than I am :)

Because of my gitk issue, wanted to have some confirmation it's still working with this one, at least @augiedoggie said it still worked there.

**Comment by augiedoggie** [comment]
> Because of my gitk issue, wanted to have some confirmation it's still working with this one, at least @augiedoggie said it still worked there.

I haven't tried gitk, only minimal testing of the main git commands

**Comment by Begasus** [comment]
> I haven't tried gitk, only minimal testing of the main git commands

My bad, I thought you did :)

**Comment by Begasus** [comment]
Hold off for a bit, doing some changes in tk (for tkinter)

**Comment by Begasus** [comment]
Since tcltk has been merged this one is good to go if no one objects

**Pull request merged**